lib/rides/pages/search_ride_page.dart

```diff
@@ -125,133 +105,270 @@ class _SearchRideFormState extends State<SearchRideForm> {
+      child: TextFormField(
+        decoration: InputDecoration(
+          border: const OutlineInputBorder(),
+          labelText: S.of(context).formTime,
+          errorText: _dateTimeValidator(_timeController.text),
         ),
+        readOnly: true,
+        onTap: _showTimePicker,
+        controller: _timeController,
+        validator: _dateTimeValidator,
       ),
     );
   }
 
-  Widget buildSeatsPicker() {
-    return Expanded(
+  Widget buildMainContentSliver() {
+    if (_rideSuggestions == null || _loading) {
+      if (_startSuggestion == null || _destinationSuggestion == null) {
+        return SliverToBoxAdapter(
+          child: Column(
+            children: <Widget>[
+              Image.asset('assets/pointing_up.png'),
+              Text(
+                S.of(context).pageSearchRideNoInput,
+                style: Theme.of(context).textTheme.headlineMedium,
+                textAlign: TextAlign.center,
+              ),
+            ],
+          ),
+        );
+      }
+      return const SliverToBoxAdapter(
+        child: Padding(
+          padding: EdgeInsets.only(top: 10),
+          child: Center(
+            child: CircularProgressIndicator(),
+          ),
+        ),
+      );
+    }
+    List<Ride> filteredSuggestions = _filter.apply(_rideSuggestions!, _selectedDate);
+    if (filteredSuggestions.isEmpty) {
+      return SliverToBoxAdapter(
+        child: SingleChildScrollView(
+          physics: const AlwaysScrollableScrollPhysics(),
+          child: Column(
+            children: <Widget>[
+              Image.asset('assets/shrug.png'),
+              Text(
+                S.of(context).pageSearchRideEmpty,
+                style: Theme.of(context).textTheme.headlineMedium,
+                textAlign: TextAlign.center,
+              ),
+              if (_rideSuggestions!.isNotEmpty)
+                Semantics(
+                  button: true,
+                  tooltip: S.of(context).pageSearchRideTooltipFilter,
+                  child: InkWell(
+                    onTap: () => _filter.dialog(context, setState),
+                    child: Text(
+                      S.of(context).pageSearchRideRelaxRestrictions,
+                      style: Theme.of(context).textTheme.titleMedium,
+                      textAlign: TextAlign.center,
+                    ),
+                  ),
+                )
+              else
+                Text(
+                  S.of(context).pageSearchRideNoResults,
+                  style: Theme.of(context).textTheme.titleMedium,
+                  textAlign: TextAlign.center,
+                ),
+            ],
+          ),
+        ),
+      );
+    } else {
+      return SliverList(
+        delegate: SliverChildBuilderDelegate((BuildContext context, int index) {
+          final Ride ride = filteredSuggestions[index];
+          return RideCard(ride);
+        }, childCount: filteredSuggestions.length),
+      );
+    }
+  }
+
+  Widget buildDateRow() {
```

Comment:
I love the lokk and the way you set it up. I don't feels so good about The Date, Seats and Filter being attached to the Scrolleable at the top. Couldn't we like make it a DropDown and hide it the moment we start scrolling?

Reply:
Do you mean that it should immediately show again the moment you start scrolling up like the AppBar? I agree, but that is incredibly complicated to implement and I would like to leave it for now

Reply:
Its ok, we can add it later then if we have time

Reply:
Time? what is that? 🤣